app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -0,0 +1,61 @@
+module ManageIQ::Providers
+  class Hawkular::MiddlewareManager < ManageIQ::Providers::MiddlewareManager
+    require_nested :RefreshParser
+    require_nested :RefreshWorker
+    require_nested :Refresher
+
+    include AuthenticationMixin
+
+    DEFAULT_PORT = 80
+    default_value_for :port, DEFAULT_PORT
+    has_many :middleware_servers, :foreign_key => :ems_id, :dependent => :destroy
+    has_many :middleware_deployments, :foreign_key => :ems_id, :dependent => :destroy
+
+    def verify_credentials(auth_type = nil, _options = {})
+      auth_type || 'default'
+    end
+
+    def self.raw_connect(hostname, port, username, password)
+      require 'hawkular_all'
+      url = URI::HTTP.build(:host => hostname, :port => port.to_i, :path => '/hawkular/inventory').to_s
+      ::Hawkular::Inventory::InventoryClient.new(url, :username => username, :password => password)
+    end
+
+    def connect
+      $log.debug("Connecting to Hawkular on #{hostname}:#{port}")
+      @connection ||= self.class.raw_connect(hostname,
```

Comment:
I know I did this `@connection` thing for the POC of the code, but it should not be here.  Instead, connect should just return the handle directly, and the connection method down below should not exist.  If a connection is needed elsewhere in the code, you should get a `with_connection` method for free by being a subclass of ContainerManager.  Helper methods like feeds, eaps, and children should probably be in the refresh code directly *or* create a helper class that delegates to the ::Hawkular::Inventory::InventoryClient handle.  children in particular is a bad method here as it shadows the methods from the RelationshipMixin

Additionally, the connect method (I thought?) took some options.  See a different providers implementation for examples (probably infra_manager/redhat is a good one)

If you need help with this part, let me know and we can pair program for a little bit.